### PR TITLE
refactor: kafka sink to serialize message as protobuf

### PR DESCRIPTION
**Pull request by kushsharma**
Converting kafka sink to serialize messages to proto by default instead of json
- for now this sink doesn't push kafka key, only value as the proto message

### CODE REVIEW

File: plugins/sinks/console/sink.go [hunk 1]
```diff
@@ -25,13 +25,12 @@ func (s *Sink) Sink(ctx context.Context, config map[string]interface{}, out <-ch
 }
 
 func (s *Sink) Process(value interface{}) error {
-	data, err := json.Marshal(value)
+	jsonBytes, err := json.Marshal(value)
 	if err != nil {
 		fmt.Println(fmt.Sprint(value))
```
Comment (StewartJingga) [inline plugins/sinks/console/sink.go:30]: Btw why do we contain the error here? 🤔

Comment (kushsharma) [reply plugins/sinks/console/sink.go:30]: Haha this is not my change although I should have cleaned it up, I only renamed the variable while resolving merge conflicts. Let me fix it.

File: plugins/sinks/kafka/sink.go [hunk 2]
```diff
@@ -2,79 +2,67 @@ package kafka
...
-	}
-}
-
 func New() core.Syncer {
 	return new(Sink)
 }
 
-// Kafka sink
 func (s *Sink) Sink(ctx context.Context, config map[string]interface{}, out <-chan interface{}) (err error) {
+	if config["brokers"] == nil {
```
Comment (StewartJingga) [inline plugins/sinks/kafka/sink.go:19]: for handling config, you can use this [utility](https://github.com/odpf/meteor/blob/main/utils/config.go) to help you build, validate and use default values.

sample usage can be found here:
https://github.com/odpf/meteor/blob/main/plugins/extractors/mysql/extractor.go#L38

as for now the error message is not being output clearly which Im currently working on right now.

Comment (kushsharma) [reply plugins/sinks/kafka/sink.go:19]: Got it, using the util!

File: plugins/sinks/kafka/sink.go [hunk 3]
```diff
@@ -2,79 +2,68 @@ package kafka
...
-}
-
 func New() core.Syncer {
 	return new(Sink)
 }
 
-// Kafka sink
-func (s *Sink) Sink(ctx context.Context, config map[string]interface{}, out <-chan interface{}) (err error) {
+type Config struct {
+	Brokers string `mapstructure:"brokers"`
```
Comment (StewartJingga) [inline plugins/sinks/kafka/sink.go:20]: super sorry for not being clear, you need to define if the field if required using tag like
```
Brokers string `mapstructure:"brokers" validate: "required"`
```

Sample:
https://github.com/odpf/meteor/blob/main/plugins/extractors/mysql/extractor.go#L23

File: plugins/sinks/kafka/sink.go [hunk 4]
```diff
@@ -2,79 +2,68 @@ package kafka
...
+	Topic string `mapstructure:"topic"`
+}
 
-	producer, err := getProducer(config)
-	if err != nil {
+func (s *Sink) Sink(ctx context.Context, config map[string]interface{}, out <-chan interface{}) (err error) {
```
Comment (StewartJingga) [inline plugins/sinks/kafka/sink.go:24]: would `in <-chan interface{}` be better here?

File: plugins/sinks/kafka/sink.go [hunk 5]
```diff
@@ -2,79 +2,68 @@ package kafka
...
+	producerConf := &kafka.ConfigMap{}
+	producerConf.SetKey("bootstrap.servers", brokers)
+	producerConf.SetKey("acks", "all")
+	return kafka.NewProducer(producerConf)
 }
 
-func (s *Sink) Push(producer *kafka.Producer, deliveryChan chan kafka.Event, value interface{}, config map[string]interface{}) error {
-	data, err := json.Marshal(value)
-
+func (s *Sink) Push(producer *kafka.Producer, value interface{}, topic string) error {
```
Comment (StewartJingga) [inline plugins/sinks/kafka/sink.go:49]: should we use lowercase `push` as the method name just to quickly show that the function is private and not being used elsewhere.

**Review by StewartJingga — CHANGES_REQUESTED** [review]
add required tag on config struct to make sure config is valid.

**Review by StewartJingga — APPROVED** [review]

**Pull request merged**